js/bruh-dash.js

```diff
@@ -1,83 +1,251 @@
 var global = window || GLOBAL;
-
 global.bruhdash = {
-  chunk: function(){
-
-  },
-
-  compact: function() {
-
-  },
-
-  difference: function() {
-
-  },
-
-  drop: function(){
-
-  },
-
-  dropRight: function() {
-
-  },
-
-  fill: function() {
-
-  },
-
-  first: function () {
-      
-  },
-
-  indexOf: function () {
-
-  },
-
-  inital: function () {
-
-  },
-
-  last: function () {
-
-  },
-
-  lastIndexof: function () {
-
-  },
-
-  pull: function () {
-
-  },
-
-  pullAt: function () {
-
-  },
-
-  rest: function () {
-
-  },
-
-  slice: function () {
-
-  },
-
-  take: function () {
-
-  },
-
-  takeRight: function () {
-
-  },
-
-  zip: function () {
-
-  },
-
-  unzip: function () {
-
-  },
-
-  without: function() {
-
+  chunk: function(array, size){
+    if (size < 1 || size === null || isNaN(size)) {
+      size = 1;
+    }
+    var chunkArray = [];
+    for (var i = 0; i < array.length; i += size) {
+      chunkArray.push(array.slice(i, i + size));
+    }
+    return chunkArray;
+
+  },
+
+  compact: function(array) {
+    var compactArray = [];
+    for (var i = 0; i < array.length; i++) {
+      if (array[i]) {
+      compactArray.push(array[i]);
+      }
+    }
+    return compactArray;
+  },
+
+  difference: function(array, ignored) {
+    var differenceArray = [];
+    for (var i = 0; i < array.length; i++) {
+      var compare = false;
+      for (var j = 0; j < ignored.length; j++) {
+        if (array[i] === ignored[j]) {
+          compare = true;
```

Comment:
While I'd suggest that you should try to do this part _without_ a nested for-loop, this solution is correct.

Thouh, you might want to `break` out of the inner for-loop once you find a match. That way you can avoid pointless checks and the program will run faster.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/break
💁